AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -615,29 +612,30 @@ am_Error_e CAmSocketHandler::removeTimer(const sh_timerHandle_t handle)
 
     //stop the current timer
 #ifdef WITH_TIMERFD 
-    std::list<sh_timer_s>::iterator it = mListTimer.begin();
-    for (; it != mListTimer.end(); ++it)
+    std::list<sh_timer_s>::iterator it(mListTimer.begin());
+    while (it != mListTimer.end())
     {
         if (it->handle == handle)
-            break;
+        {
+            mListTimer.erase(it);
+            return removeFDPoll(handle);
+        }
+        ++it;
     }
-    if (it == mListTimer.end())
-        return (E_NON_EXISTENT);
+    return (E_NON_EXISTENT);
 
-    close(it->fd);
-    mListTimer.erase(it);
-    return removeFDPoll(handle);
 #else
     stopTimer(handle);
     std::list<sh_timer_s>::iterator it(mListTimer.begin());
-    for (; it != mListTimer.end(); ++it)
+    while (it != mListTimer.end())
     {
         if (it->handle == handle)
         {
-            it = mListTimer.erase(it);
+            mListTimer.erase(it);
```

Comment:
In principle you are right. But we return in this case! The iterator will be not used afterwards.

Reply:
I'm looking both for correctness, and consistency (and avoiding mistakes if someone reuses / changes / copies code).  Unifying this is something that could be done in a later cleanup.

Reply:
Yes, it will work. I like the idea. I consider to change also the order of the enum to align it better to the internal state machine:
´´´
typedef enum:uint8_t
{
    ADD     = 1u,
    UPDATE  = 2u,
    VALID   = 3u,
    CLOSE   = 4u,
    REMOVE  = 5u
´´´